docs/inventory_object.md

```diff
@@ -0,0 +1,187 @@
+# inventory directive in kustomization.yaml
+
+New in v2.1.0, a kustomization file may have an `inventory` field:
+```yaml
+inventory:
+  type: ConfigMap
+  configMap:
+    name: prune-cm-name
+    namespace: some-namespace
+```
+
+### Motivation
+If present, `Kustomize build` will make an _inventory_ object,
+which  could be a ConfigMap, or an App(to be added),
+ which can be consumed by a client such as those under development in
+[cli-experimental](https://github.com/kubernetes-sigs/cli-experimental).
+The client can recognize this object by name and use it to do a better job
+with actions like `apply`, `prune` and `delete`.
+
+
+### Implementation
+
+The _inventory_ ConfigMap contains two special annotations:
+
+- kustomize.config.k8s.io/Inventory
+  The value of this annotation is the JSON blob
+  for an Inventory object. The Inventory is a
+  struct that contains following information
+  - all objects within this kusotmization target
+  - all objects that reference within this kustomization target
+  
```

Comment:
nit: all objects that reference OR all objects that are referenced by?

Reply:
I changed it to `all object reference relationships`